### PR TITLE
Update styling to prevent bc-table-row-header text from overflowing its content-box

### LINKED ISSUE #4230: Longer method names or attributes overflow BC table cell
Related to #4176 

On the https://developer.mozilla.org/en-US/docs/Web/API/AudioContext/createMediaStreamDestination page, some of the longer method names or attributes are overflowing the cell. 

Screen Shot 2021-07-16 at 10 22 34 AM

### LINKED ISSUE #4509: Issue with "Content Security Policy (CSP)": First column of Compatability Table is truncated to be of no use.
MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP

#### What information was incorrect, unhelpful, or incomplete?
The first column of the Browser Compatibility table lists 'Content-Security-Pol' for every row.  The full text is in the DOM, but truncated.  Changing browser zoom did not help.

image
#### What did you expect to see?
Either the TD data split onto two rows, or the repeated 'Content-Security-Policy' prefix removed.

#### Did you test this? If so, how?

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/http/csp`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/http/csp/index.md
* Last commit: https://github.com/mdn/content/commit/6cecaf6b41e8b366bbb57755f85334fd0d99d55e
* Document last modified: 2021-08-14T15:42:25.000Z

</details>

### PR DESCRIPTION
**Pull request by tannerdolby**
Fixes #4230 
Fixes #4509

Screen Shot 2021-07-21 at 11 25 11 AM

### CODE REVIEW

**Comment by tannerdolby** [comment]
https://user-images.githubusercontent.com/48612525/126137584-ff5b1ef9-65ee-419d-9e92-5292ba0d6edb.mov

- Original article by Ben Nadel: https://www.bennadel.com/blog/3626-trying-to-center-a-text-overflow-ellipsis-using-css-flexbox-in-angular-7-2-15.htm
- CodePen demo: https://codepen.io/tannerdolby/pen/OJmmpXo

**Comment by tannerdolby** [comment]
cc @schalkneethling @peterbe

File: client/src/document/ingredients/browser-compatibility-table/feature-row.tsx [hunk 1]
```diff
@@ -425,6 +425,29 @@ function CompatCell({
...
+  let rightSide: string = text.slice(cutoff);
+  return (
+    <>
+      <code title={text}>
+        <span className="left-side">{leftSide}</span>
+        <span className="right-side">{rightSide}</span>
+      </code>
+    </>
+  );
+}
```
Comment (peterbe) [inline client/src/document/ingredients/browser-compatibility-table/feature-row.tsx:449]: This TS code needs some work. I can give a more detailed review if and when @schalkneethling has had a chance to review the total effect of this functionality.

Comment (tannerdolby) [reply client/src/document/ingredients/browser-compatibility-table/feature-row.tsx:449]: Yeah, looking back at this. I did define a few types, but the it definitely needs some work.

Comment (tannerdolby) [reply client/src/document/ingredients/browser-compatibility-table/feature-row.tsx:449]: Outside of the TypeScript updates. What do you think of how the functionality works? From a reader standpoint? @peterbe

Comment (peterbe) [reply client/src/document/ingredients/browser-compatibility-table/feature-row.tsx:449]: > Outside of the TypeScript updates. What do you think of how the functionality works? From a reader standpoint? @peterbe

Haven't had a chance to even look that far :)
Will do when time allows. But for now I lean on Schalk to take a first overview-look.

Comment (tannerdolby) [reply client/src/document/ingredients/browser-compatibility-table/feature-row.tsx:449]: > Haven't had a chance to even look that far :) Will do when time allows. But for now I lean on Schalk to take a first overview-look

Ok no worries :) thanks for the heads up.

**Comment by peterbe** [comment]
Is there no simple CSS fancy technique to tell the text to become smaller in font if it doesn't fit?

**Comment by peterbe** [comment]
> Is there no simple CSS fancy technique to tell the text to become smaller in font if it doesn't fit?

I say that because this is quite a few lines of code for something that's probably very rare. And that extra JS adds weight to ALL pages potentially.

**Comment by tannerdolby** [comment]
> Is there no simple CSS fancy technique to tell the text to become smaller in font if it doesn't fit?

Unfortunately, no.

**Comment by tannerdolby** [comment]
> I say that because this is quite a few lines of code for something that's probably very rare. And that extra JS adds weight to ALL pages potentially.

Yeah. After seeing the performance score in #4060 and all the client-side JS that's being loaded, the less JS we add to the page the better. I will submit some changes here to see if this could be converted to cheerio, so the CSS is what drives the change with cheerio (instead of javascript).

**Comment by tannerdolby** [comment]
Honestly, @peterbe I think your right. Although the "sliding-door" is pretty cool. It might be overkill and probably should only be used a last resort, tried with cheerio but we can't access the stuff in the `.bc-table-row-header` with an `each` loop which is a shame. 

One "pure css" solution, without bothering with `text-overflow: ellipsis` would be to simply add `white-space: break-spaces` or `white-space: normal` for something to break the text to next line and handle how white space is collapsed. This seems so quite obvious and doesn't cause any immediate a11y issues from my understanding.

```css
.bc-table-row-header code {
    display: flex;
    overflow: hidden;
    white-space: normal;
}
```

Updated:

_When cell is small and text would otherwise overflow_
Screen Shot 2021-07-21 at 10 59 34 AM

_When cell is large and text overflow is not likely_
Screen Shot 2021-07-21 at 10 59 44 AM

**Comment by tannerdolby** [comment]
It works smoothly. But I noticed one hiccup, for the text that is a link.

Screen Shot 2021-07-21 at 11 22 22 AM

Can be resolved by making `.bc-table-row-header` a flexbox with `display: flex`

Screen Shot 2021-07-21 at 11 25 11 AM

**Comment by schalkneethling** [comment]
Thank you for this @tannerdolby. I know @dalexsy is also looking at these tables as part of the redesign so, perhaps this is something we will address as part of the overall changes.

**Comment by tannerdolby** [comment]
> Thank you for this @tannerdolby. I know @dalexsy is also looking at these tables as part of the redesign so, perhaps this is something we will address as part of the overall changes.

Your welcome @schalkneethling. Ok, that works for me! I definitely think this `white-space: normal` route is worth exploring as it doesn't create a lot of fuss while keeping the text visible for assistive technology and sighted users. I will stay tuned as this is addressed as part of the overall site changes.

**Comment by peterbe** [comment]
@schalkneethling Will you own this from now on? Or is it still valueable for me to review?

**Comment by schalkneethling** [comment]
> @schalkneethling Will you own this from now on? Or is it still valueable for me to review?

I can take it.

**Review by schalkneethling — APPROVED** [review]
r+ 🎉

**Pull request merged**